### PR TITLE
EVEREST-107 Upgrade olm template

### PR DESCRIPTION
**Pull request by oksana-grishchenko**
Add support for the new basic template (previously the schemas were a separate docs, and now they are in a single doc placed into the `Entries` array) :

- update the generation tool to use Go structures
- update tests
- remove the `releaseType` parameter

### CODE REVIEW

File: tools/veneer-update.go [hunk 1]
```diff
@@ -61,215 +46,110 @@ func main() {
...
 
-		// Add current version to skips
-		skips, ok := newEntry["skips"]
-		if !ok {
-			skips = []string{}
-		}
+type Template struct {
+	Schema  string  `yaml:"schema"`
+	Entries []Entry `yaml:"entries"`
+}
```
Comment (recharte) [inline tools/veneer-update.go:61]: Should we import the modules and use the official structs instead?
Your `Template` is the  [BasicTemplate](https://github.com/operator-framework/operator-registry/blob/3be1d97c5d8dfe672ba66e7b8fd4aaaa21fe7d2b/alpha/template/basic/basic.go#L21-L24) and the `Entry` is the [DeclarativeConfig and child structs](https://github.com/operator-framework/operator-registry/blob/3be1d97c5d8dfe672ba66e7b8fd4aaaa21fe7d2b/alpha/declcfg/declcfg.go#L25-L60)

File: tools/veneer-update.go [hunk 2]
```diff
@@ -61,215 +46,110 @@ func main() {
...
+type Version struct {
+	Name     string   `yaml:"name"`
+	Replaces string   `yaml:"replaces,omitempty"`
+	Skips    []string `yaml:"skips,omitempty"`
+}
 
-		newEntry["skips"] = append(skipsSlice, fmt.Sprintf("everest-operator.v%s", currentVersion))
+const (
+	olmChannelSchema = "olm.channel"
+	olmBundleSchema  = "olm.bundle"
```
Comment (recharte) [inline tools/veneer-update.go:71]: https://github.com/operator-framework/operator-registry/blob/3be1d97c5d8dfe672ba66e7b8fd4aaaa21fe7d2b/alpha/declcfg/declcfg.go#L20-L21

File: tools/everest_basic_template.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	if err != nil {
+		return err
+	}
+
+	jsonData, err := json.Marshal(yamlData)
+	if err != nil {
+		return err
+	}
+
+	return json.Unmarshal(jsonData, t)
```
Comment (recharte) [inline tools/everest_basic_template.go:35]: Is this JSON marshalling/unmarshalling really needed? Couldn't we directly unmarshal the yaml?

File: tools/veneer-update.go (hunk 1 above)
Comment (oksana-grishchenko) [reply tools/veneer-update.go]: Thanks for highlighting this! Updated and refactored. 

Advantages:
1. Consistent structs. We know for sure that the structs we use reflect the basic template structure.

Disadvantages: 
1.  more code bc: 
- the structs doesn't have the yaml tags so we need to handle the json-yaml transformations 
- marshalling and unmarshalling blobs inside the structure
3. A big number of new indirect dependencies

In general the advantage may worth the disadvantages. 
Considering the refactoring, the current implementation looks solid and maintainable.

File: tools/everest_basic_template.go (hunk 3 above)
Comment (oksana-grishchenko) [reply tools/everest_basic_template.go:35]: no we can't, bc the original structure doesn't have the yaml tags and can't work with the blob. so we first need to go through the json and create the proper struct.

**Review by recharte — APPROVED** [review]
Good job :muscle:

**Pull request merged**